genapi/genapi.go

```diff
@@ -881,6 +894,21 @@ func (g *GenAPI) RequestContext(r *http.Request) context.Context {
 	return ctx
 }
 
+// ReloadListeners stops all the existing listeners and remakes them. Only
+// useful in APIMode.
+func (g *GenAPI) ReloadListeners() {
+	// first close all the previous listeners
+	for _, l := range g.listeners {
+		log.Printf("closing %v", l)
+		l.Close()
```

Comment:
Might want some timeout on this... like if its takes more than x seconds just force kill it.